Modules/Core/Common/include/itkImageRandomNonRepeatingConstIteratorWithIndex.h

```diff
@@ -324,10 +324,20 @@ class ITK_TEMPLATE_EXPORT ImageRandomNonRepeatingConstIteratorWithIndex : public
 
   /** Set/Get number of random samples to extract from the image region. */
   void
-  SetNumberOfSamples(SizeValueType number);
+  SetNumberOfSamples(SizeValueType number)
+  {
+    m_NumberOfSamplesRequested = number;
+    if (number > m_NumberOfPixelsInRegion)
+    {
+      m_NumberOfSamplesRequested = m_NumberOfPixelsInRegion;
+    }
```

Comment:
Looks like `m_NumberOfSamplesRequested = std::min(number, m_NumberOfPixelsInRegion)` (but that's beyond the scpe of this PR anyway, of course).